src/test/ui/mir/issue-66930.rs

```diff
@@ -0,0 +1,11 @@
+// check-pass
+// compile-flags: --emit=mir,link
```

Comment:
Ah nice, I wondered what would be a good way to test this. :)

Is this compatible with `check-pass`, or does that somehow make it so that we do not actually emit any MIR?

Reply:
Does "this" means the `link` flag? If so, we can avoid linking failure by adding this.

Reply:
I figured `check-pass` would also play with `--emit` so also setting that in compile-flags might not have any effect. And since you added `link` isn't this effectively `build-pass` anyway?

Reply:
I used them since other tests that have `--emit=mir` are also with `check-pass`. I think if we use `check-pass`, then we need `link` flag as well, if we use `build-pass`, then we don't need it. It's just an option imo.

Reply:
Could you verify that this is actually true? Is this documented somewhere? Would be a shame to add a test that doesn't actually test anything.

My suspicion is still that the `check-pass`+`--emit=link` is the same as `build-pass` so it's a bit useless... except maybe it makes a difference when we also have `--pass check`?

Reply:
> Could you verify that this is actually true?

Sure, I checked out f93bb2a50b37bc8bafe4d960e2afd839eaa854ed and run the test, it emitted ICE expectedly either `check-pass` with `link` or `build-pass`.

> Is this documented somewhere?

I guess it isn't.

So if we prefer `build-pass` here for simplify then I'll update so.

Reply:
> Sure, I checked out f93bb2a and run the test, it emitted ICE expectedly either check-pass with link or build-pass.

Thanks!